src/main/java/com/example/bank/bankTransfer/transfer/history/TransferHistoryService.java

```diff
@@ -0,0 +1,106 @@
+package com.example.bank.bankTransfer.transfer.history;
+
+import com.example.bank.bankTransfer.account.Account;
+import com.example.bank.bankTransfer.account.AccountRepository;
+import com.example.bank.bankTransfer.transfer.TransferType;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.apache.poi.ss.usermodel.DataFormat;
+import org.apache.poi.ss.usermodel.FillPatternType;
+import org.apache.poi.xssf.usermodel.*;
+import org.springframework.stereotype.Service;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+@Getter
+public class TransferHistoryService {
+    private final AccountRepository accountRepository;
+    private final TransferHistoryRepository transferHistoryRepository;
+
+    public List<TransferHistory> transferHistoryForClient(Integer clientId) {
+        return transferHistoryRepository.findByClientId(clientId);
+    }
+
+    public TransferHistory buildFromAccountTransferHistory(Account fromAccount, BigDecimal amount, Account toAccount) {
+        return TransferHistory.builder()
+                .transferType(TransferType.EXPENSE)
+                .clientId(fromAccount.getClient().getId())
+                .beforeBalance(fromAccount.getBalance())
+                .afterBalance(fromAccount.getBalance())
+                .amount(amount)
+                .bankAccountId(toAccount.getId())
+                .build();
+    }
+
+    public TransferHistory buildToAccountTransferHistory(Account toAccount, BigDecimal amount, Account fromAccount) {
+        return TransferHistory.builder()
+                .transferType(TransferType.INCOME)
+                .clientId(toAccount.getClient().getId())
+                .beforeBalance(toAccount.getBalance())
+                .afterBalance(toAccount.getBalance())
+                .amount(amount)
+                .bankAccountId(fromAccount.getId())
+                .build();
+    }
+
+    //TODO:  decide whether clientId or accountId for bank statement
+    public void generateXlsxBankStatement(Integer clientId) {
```

Comment:
czy przez dopracowac miales na mysli napisac to od nowa tak jak powinno to byc napisane? Zgodnie z zasadami SOLID DRY KISS ? itd?

Reply:
W sumie nie. Chodziło mi o weryfikację kodu pod kątem transakcji. 
Niemniej jednak w myśl zasady YAGNI usunąłem klasę (encję) Transfer, bo nie jest na razie potrzebna i raczej już nie będzie.

Reply:
Sam przelew oprocz komentarzy ktore napisalem wyglada okej. oczywiscie testy trzeba napisac bo chyba ich tam nie widzialem